### DISCUSSION TITLE
Nothing visible in the displayer when trying to addGeometry from widget, but works when I do addGeometry from window.

### DISCUSSION THREAD

**Discussion by MeyerBuaharon** [isl-org/Open3D#4500] [Q&A]
I try to make in cpp a running displayer that runs on a thread and always redraw plys. I did it in python and it worked perfectly, but I need to recreate the code in cpp, but im having a bit of difficulty with the widget.

`int main(int argc, char *argv[]) {
    int width = 1024;
    int height = 768;
    std::string window_name = "Open3D";
    Eigen::Vector3f center = {0.0, 0.0, 0.0};
    Eigen::Vector3f eye = {125.0, 125.0, 125.0};
    Eigen::Vector3f up = {0.0, 0.0, 1.0};

    gui::Application::GetInstance().Initialize();

    auto window =
            std::make_shared<open3d::visualization::visualizer::O3DVisualizer>(
                    window_name, width, height);
                    auto *scene = new rendering::Open3DScene();
    auto widget = std::make_shared<gui::SceneWidget>();
     
    widget->SetScene(
            std::make_shared<rendering::Open3DScene>(window->GetRenderer()));
    widget->EnableSceneCaching(true);  // smoother UI with large geometry
    window->AddChild(widget);
    cout << window->GetScene() << endl;

    gui::Application::GetInstance().AddWindow(window);

    auto mat = rendering::MaterialRecord();

    mat.shader = "defaultLit";

    window->GetScene()->GetCamera()->LookAt(center, eye, up);

    std::shared_ptr<geometry::PointCloud> pointcloud_ptr_A(
            new geometry::PointCloud);

    geometry::PointCloud pointcloudA;

    io::ReadPointCloud("../SplitCube/" + std::to_string(1) + "A.ply",
                       pointcloudA);

    *pointcloud_ptr_A = pointcloudA;

    widget->GetScene()->AddGeometry("pcdA", pointcloud_ptr_A.get(), mat);
        thread t([window, widget]() { run(window, widget); });
        t.detach();
    gui::Application::GetInstance().Run();

    return 0;
}`

this is my main, and as you can see when I try to do AddGeometry to pointcloud_ptr_A I get a blank displayer, but when I try and do `window->GetScene()->AddGeometry(...)` this show the ply.

Can anyone here point out what I did wrong?
Thanks for this tool, it is really interesting!
